src/subcommand/wallet/inscribe/batch.rs

```diff
@@ -600,9 +611,20 @@ impl Batchfile {
         compress,
       )?);
 
+      if !(self.mode == Mode::SharedOutput && i >= 1) {
```

Comment:
Maybe try refactoring this into something like this, so that we get some type safety:

```rust
enum Destinations {
  Shared { inscriptions: Vec<Inscription>, destination: Address },
  Separate { bar: Vec<(Inscription, Address)> }
}

impl InscriptionDestinations {
  fn bar(&self) -> Vec<(Inscription, Address)> {
    match self {
      Self::Shared { inscription, destination} => {
        self.inscriptions.map(|inscription| (inscription, self.address)).collect(),
      }
      Self::Separate { bar} => self.bar.iter().collect(),
    }
  }
}
```